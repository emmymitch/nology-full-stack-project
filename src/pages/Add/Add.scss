@use "../../styles/variables" as *;

.add{
    display: flex;
    flex-direction: column;
    align-items: center;

    & > * {
        margin: 1rem 0;
    }

    &__form{
        &--style-div{
            display: flex;
            flex-direction: column;
            align-items: center;
        }

        &--label{
            margin: 0.75rem 0 0.25rem 0;
        }

        &--domain-list, &--identifier-list{
            display: flex;
            flex-direction: column;
        }

        &--description{
            width: 20rem;
            height: 5rem;
            font-size: 0.85rem;
        }
    }
}