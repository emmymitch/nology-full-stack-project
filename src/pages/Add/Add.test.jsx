import Add from "./Add";
import { render } from "@testing-library/react";

it("Should render the Add page", () => {
  const container = render(<Add />)

  expect(container).toMatchSnapshot();
});