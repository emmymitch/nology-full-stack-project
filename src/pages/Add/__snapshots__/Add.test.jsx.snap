// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Add page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="add"
      >
        <h1
          class="add__title"
        >
          Add
        </h1>
        <h3
          class="add__subheading"
        >
          Missing someone? Add them to the database here!
        </h3>
        <form
          class="add__form"
        >
          <button
            class="button button--reset"
            type="reset"
          >
            Reset
          </button>
          <br />
          <div
            class="add__form--style-div"
          >
            <label
              class="add__form--label"
              for="createdBy"
            >
              Author:
            </label>
            <input
              class="text-input "
              name="createdBy"
              placeholder="Who's authoring this entry?"
              type="text"
              value=""
            />
            <br />
            <label
              class="add__form--label"
              for="englishName"
            >
              English Name:
            </label>
            <input
              class="text-input "
              name="englishName"
              placeholder=""
              type="text"
              value=""
            />
            <br />
            <label
              class="add__form--label"
              for="greekName"
            >
              Greek Name:
            </label>
            <input
              class="text-input "
              name="greekName"
              placeholder="Use Greek characters only"
              type="text"
              value=""
            />
            <br />
            <label
              class="add__form--label"
              for="majorDomains"
            >
              Major Domains:
            </label>
            <div
              class="add__form--domain-list"
            >
              <input
                class="text-input "
                name="majorDomains"
                placeholder=""
                type="text"
                value=""
              />
              <input
                class="text-input "
                name="majorDomains"
                placeholder=""
                type="text"
                value=""
              />
              <input
                class="text-input "
                name="majorDomains"
                placeholder=""
                type="text"
                value=""
              />
              <input
                class="text-input "
                name="majorDomains"
                placeholder=""
                type="text"
                value=""
              />
              <input
                class="text-input "
                name="majorDomains"
                placeholder=""
                type="text"
                value=""
              />
            </div>
            <br />
            <label
              class="add__form--label"
              for="identifiers"
            >
              Identifiers:
            </label>
            <div
              class="add__form--identifier-list"
            >
              <input
                class="text-input "
                name="identifiers"
                placeholder="e.g. lightning bolts and eagles for Zeus"
                type="text"
                value=""
              />
              <input
                class="text-input "
                name="identifiers"
                placeholder="e.g. owls, olives, Aegis for Athena"
                type="text"
                value=""
              />
              <input
                class="text-input "
                name="identifiers"
                placeholder="e.g. lyre, bow, laurel for Apollo"
                type="text"
                value=""
              />
              <input
                class="text-input "
                name="identifiers"
                placeholder="e.g. a caduceus and winged sandals for Hermes"
                type="text"
                value=""
              />
              <input
                class="text-input "
                name="identifiers"
                placeholder="e.g. doves and roses for Aphrodite"
                type="text"
                value=""
              />
            </div>
            <br />
            <label
              class="add__form--label"
              for="description"
            >
              Description:
            </label>
            <textarea
              class="add__form--description"
              name="description"
              placeholder="A short description of the god. Exclude personal opinions."
            />
            <br />
            <button
              class="button "
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="add"
    >
      <h1
        class="add__title"
      >
        Add
      </h1>
      <h3
        class="add__subheading"
      >
        Missing someone? Add them to the database here!
      </h3>
      <form
        class="add__form"
      >
        <button
          class="button button--reset"
          type="reset"
        >
          Reset
        </button>
        <br />
        <div
          class="add__form--style-div"
        >
          <label
            class="add__form--label"
            for="createdBy"
          >
            Author:
          </label>
          <input
            class="text-input "
            name="createdBy"
            placeholder="Who's authoring this entry?"
            type="text"
            value=""
          />
          <br />
          <label
            class="add__form--label"
            for="englishName"
          >
            English Name:
          </label>
          <input
            class="text-input "
            name="englishName"
            placeholder=""
            type="text"
            value=""
          />
          <br />
          <label
            class="add__form--label"
            for="greekName"
          >
            Greek Name:
          </label>
          <input
            class="text-input "
            name="greekName"
            placeholder="Use Greek characters only"
            type="text"
            value=""
          />
          <br />
          <label
            class="add__form--label"
            for="majorDomains"
          >
            Major Domains:
          </label>
          <div
            class="add__form--domain-list"
          >
            <input
              class="text-input "
              name="majorDomains"
              placeholder=""
              type="text"
              value=""
            />
            <input
              class="text-input "
              name="majorDomains"
              placeholder=""
              type="text"
              value=""
            />
            <input
              class="text-input "
              name="majorDomains"
              placeholder=""
              type="text"
              value=""
            />
            <input
              class="text-input "
              name="majorDomains"
              placeholder=""
              type="text"
              value=""
            />
            <input
              class="text-input "
              name="majorDomains"
              placeholder=""
              type="text"
              value=""
            />
          </div>
          <br />
          <label
            class="add__form--label"
            for="identifiers"
          >
            Identifiers:
          </label>
          <div
            class="add__form--identifier-list"
          >
            <input
              class="text-input "
              name="identifiers"
              placeholder="e.g. lightning bolts and eagles for Zeus"
              type="text"
              value=""
            />
            <input
              class="text-input "
              name="identifiers"
              placeholder="e.g. owls, olives, Aegis for Athena"
              type="text"
              value=""
            />
            <input
              class="text-input "
              name="identifiers"
              placeholder="e.g. lyre, bow, laurel for Apollo"
              type="text"
              value=""
            />
            <input
              class="text-input "
              name="identifiers"
              placeholder="e.g. a caduceus and winged sandals for Hermes"
              type="text"
              value=""
            />
            <input
              class="text-input "
              name="identifiers"
              placeholder="e.g. doves and roses for Aphrodite"
              type="text"
              value=""
            />
          </div>
          <br />
          <label
            class="add__form--label"
            for="description"
          >
            Description:
          </label>
          <textarea
            class="add__form--description"
            name="description"
            placeholder="A short description of the god. Exclude personal opinions."
          />
          <br />
          <button
            class="button "
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
